src/lib/support/Span.h

```diff
@@ -42,6 +45,17 @@ class Span
     constexpr explicit Span(T (&databuf)[N]) : Span(databuf, N)
     {}
 
+    // Allow implicit construction from a Span over a type that matches our
+    // type, up to const-ness.
+    template <class U, typename = std::enable_if_t<std::is_same<std::remove_const_t<T>, std::remove_const_t<U>>::value>>
```

Comment:
```suggestion
    template <class U, typename = std::enable_if_t<std::is_convertible<U*, T*>::value>>
```

Reply:
No, this does not do the right thing.  In particular, `is_convertible<U*, T*>` will test true if `U` is a subclass of `T` with public inheritance.  So it can test true when `U` and `T` have different sizes.

So say we have a `T` that is 4 bytes and `U` inherits from it and is 8 bytes.  We have a `Span<U>` of size 2 (so holding 2 `U` instances).  We try to initialize a `Span<T>` from it, and with the `is_convertible` approach that would work and create a span of size 2, claiming to hold 2 `T` instances.  The first `T*` will point to the first `U` instance.  The second `T*` will  point to the second half of the first `U` instance, which is not right at all.